grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -144,7 +145,12 @@ class LocationDataService {
         }
         def locationType = null
         if (params.locationType) {
-            locationType = LocationType.findByIdOrName(params.locationType, params.locationType)
+            String locationTypeName = LocalizationUtil.getDefaultString(params.locationType as String)
+            // TODO: Replace with a single GORM .find with Closure when in Grails 3 (available since Grails 2.0)
+            LocationType matchedLocationType = LocationType
+                    .findAllByNameLike(locationTypeName + "%")
+                    .find{ LocalizationUtil.getDefaultString(it.name) == locationTypeName}
+            locationType = matchedLocationType
```

Comment:
Plus why you removed the option to upload a document with location type id instead of name? I think this should be also possible.

Reply:
@awalkowiak sure, this change was discussed on the tech huddle, but since you were not here, I'll try to explain it in a nutshell:

1. The load data template has `locationType` value without pipe-delimited translation, e.g. `Depot`. 
With the previous version of the code, so `findByIdOrName` it did not match the locationType, because `Depot !== Depot|fr:D`. I was advised not to change the template (so in the template from `Depot` to `Depot|fr:D` if there is any other possible way to solve the problem.

Logic behind my change:
If we were in at least Grails 2.0, we could just use GORM's `.find`, so 
```groovy
LocationType.find { LocalizationUtil.getDefaultString(it.name) == locationTypeName }
```
but this method is not yet available in Grails 1, so at first I fetch `.findAllByNameLike` not to do `.findAll`, so I try to limit the records returned in the List (not to cause unexpected performance issues), and then I do `.find`, but this `.find` is just a method on **list** (not the GORM's method) to match the locationType name with the given name in the template.
I use `LocalizationUtil.getDefaultString` method to allow users typing `Depot` in the template (before they had to type explicitly `Depot|fr:D` for the template to be imported successfully).

Let me know if it clarifies a bit

Reply:
You might say why can't I use pure SQL then if GORM's `.find` is not available in Grails 1. The reason for that is you can't say: `SELECT LocalizationUtil.getDefaultString(name) from location_type` <= you can't call a method while selecting, so that's why I had to do this "stupid" `.findAllByNameLike` and then `.find`.
@jmiranda has said that it was possible to do it via `formula` in the domain's mapping, but he was not a big fan of it.

Reply:
OK, I get it, thanks :)